qiime2_pipeline.bash

```diff
@@ -0,0 +1,177 @@
+## These can be downloaded from https://data.qiime2.org/2017.9/common/gg-13-8-99-nb-classifier.qza (full length)
+## or https://data.qiime2.org/2017.9/common/gg-13-8-99-515-806-nb-classifier.qza (515F/806R region)
+
+CLASSIFIER_FP="${HOME}/gg-13-8-99-nb-classifier.qza"
+#CLASSIFIER_FP="${HOME}/gg-13-8-99-515-806-nb-classifier.qza" ## used for V4 region
+
+DATA_DIR="${WORK_DIR}/data_files"
+
+PROJECT_DIR="${WORK_DIR}/${PROJECT_NAME}"
+if [ ! -d ${PROJECT_DIR} ]; then
+	mkdir ${PROJECT_DIR}
+fi
+
+FORMATTED_LANE_NUM=$(printf "%03d" ${LANE_NUM})
+FWD="${DATA_DIR}/Undetermined_S0_L${FORMATTED_LANE_NUM}_R1_001.fastq.gz"
+REV="${DATA_DIR}/Undetermined_S0_L${FORMATTED_LANE_NUM}_R2_001.fastq.gz"
+IDX="${DATA_DIR}/Undetermined_S0_L${FORMATTED_LANE_NUM}_I12_001.fastq.gz"
+
+###=====================
+### DATA IMPORT
+###=====================
+
+EMP_PAIRED_END_SEQUENCES_DIR="${PROJECT_DIR}/emp-paired-end-sequences"
+if [ ! -d ${EMP_PAIRED_END_SEQUENCES_DIR} ]; then
+        mkdir ${EMP_PAIRED_END_SEQUENCES_DIR}
+fi
+
+cp ${FWD} "${EMP_PAIRED_END_SEQUENCES_DIR}/forward.fastq.gz"
+cp ${REV} "${EMP_PAIRED_END_SEQUENCES_DIR}/reverse.fastq.gz"
+cp ${IDX} "${EMP_PAIRED_END_SEQUENCES_DIR}/barcodes.fastq.gz"
+
+qiime tools import \
+  --type EMPPairedEndSequences \
+  --input-path $EMP_PAIRED_END_SEQUENCES_DIR \
+  --output-path "$PROJECT_DIR/emp-paired-end-sequences.qza"
+
+###=====================
+### DEMULTIPLEXING SEQUENCE
+###=====================
+
+qiime demux emp-paired \
+  --m-barcodes-file ${MAPPING_FP} \
+  --m-barcodes-category BarcodeSequence \
+  --i-seqs "${PROJECT_DIR}/emp-paired-end-sequences.qza" \
+  --o-per-sample-sequences "${PROJECT_DIR}/demux.qza" \
+  --p-rev-comp-mapping-barcodes ## need discussion here (careful with EOF above)
+
+qiime demux summarize \
+  --i-data "${PROJECT_DIR}/demux.qza" \
+  --o-visualization "${PROJECT_DIR}/demux.qzv"
+
+qiime tools export \
+  "${PROJECT_DIR}/demux.qzv" \
+  --output-dir "${PROJECT_DIR}/demux"
+
+###=====================
+###  SEQUENCE QC AND FEATURE TABLE
+###=====================
+
+## discussion needed for denosing parameters below
+
+qiime dada2 denoise-paired \
+  --i-demultiplexed-seqs "${PROJECT_DIR}/demux.qza" \
+  --p-trim-left-f 0 \
+  --p-trunc-len-f 230 \
+  --p-trim-left-r 0 \
+  --p-trunc-len-r 230 \
+  --p-n-threads 8 \
+  --o-representative-sequences "${PROJECT_DIR}/rep-seqs.qza" \
+  --o-table "${PROJECT_DIR}/table.qza"
+
+qiime feature-table summarize \
+  --i-table "${PROJECT_DIR}/table.qza" \
+  --o-visualization "${PROJECT_DIR}/table.qzv" \
+  --m-sample-metadata-file ${MAPPING_FP}
+
+qiime feature-table tabulate-seqs \
+  --i-data "${PROJECT_DIR}/rep-seqs.qza" \
+  --o-visualization "${PROJECT_DIR}/rep-seqs.qzv"
+
+qiime tools export \
+  "${PROJECT_DIR}/table.qzv" \
+  --output-dir "${PROJECT_DIR}/table"
+
+qiime tools export \
+  "${PROJECT_DIR}/table.qza" \
+  --output-dir "${PROJECT_DIR}/table"
```

Comment:
Must clarify what the output is from `table.qzv` vs. `table.qza`

Reply:
Exporting table.qza gives feature-table.biom, while table.qzv gives a bunch of files including sample-frequency-detail.csv. Hi quality read counts will be calculated from feature-table.biom, so the step for exporting table.qzv will be removed.